device/alpaka/include/traccc/alpaka/utils/definitions.hpp

```diff
@@ -11,15 +11,40 @@
 #include <alpaka/example/ExampleDefaultAcc.hpp>
```

Comment:
I guess this is the smoking gun... :thinking: Alpaka headers should not be included in the public headers of this library. This `definitions.hpp` file could just be moved under `device/alpaka/src/`, right?

Reply:
Looking over my code, I was closely mimicing the structure of the CUDA code, which has the `#include <cuda.h>` etc in its `include/utils/definitions.hpp`, but they are obviously causing a lot more issues here due to the header-only nature of Alpaka.

I've refactored out basically all of this file into the `src/` folder, leaving only the warpSize definition and the non-alpaka definitions. That seems to have made things much happier, so I've also dropped the `alpaka_test/executable` CMake functions (as we seemingly don't currently need them).